edu-1481-guide-github-autodeploy-template (#349)
aziontech/docs: src/content/docs/pt-br/pages/guias/marketplace/templates/edge-function-github-autodeploy/edge-function-github-autodeploy.mdx

---
title: Como usar o template Edge Function GitHub AutoDeploy

> description: Este template permite implantar edge functions integradas ao Github Actions diretamente no edge.

Reviewer: ```suggestion
description: Este template permite implantar edge functions integradas ao GitHub Actions diretamente no edge.
```

meta_tags: templates, guias, Azion Marketplace
namespace: docs_guides_edge_function_github_autodeploy
permalink: /documentacao/produtos/guias/edge-function-github-autodeploy/
---

import Badge from '~/components/Badge.astro';

<Badge variant="accent">
Preview
</Badge>

> Azion **Edge Function GitHub AutoDeploy** é um template que permite implantar edge functions integradas ao Github Actions diretamente no edge da rede.

Reviewer: ```suggestion
Azion **Edge Function GitHub AutoDeploy** é um template que permite implantar edge functions integradas ao GitHub Actions diretamente no edge da rede.
```

> Ao usar este template, você só precisa criar sua edge function junto com uma aplicação, e a implantação contínua (continuous deployment) é feita por meio de GitHub Actions. Isto acelera os seus fluxos de trabalho e garante a atualização constante do seu projeto.

Reviewer: ```suggestion
Ao usar este template, você só precisa criar sua edge function junto com uma aplicação, e a implantação contínua (continuous deployment) é feita por meio de GitHub Actions. Isso acelera os seus fluxos de trabalho e garante a atualização constante do seu projeto.
```

---

> ## Obter o template

Author: ```suggestion
## Obtenha o template
```

Para começar a usar o template **Edge Function GitHub AutoDeploy**, siga os passos:

1. Acesse o [Real-Time Manager (RTM)](https://manager.azion.com/).

> - Se você ainda não tem uma conta, crie uma [visitando a página de sign up](https://manager.azion.com/signup/).

Reviewer: ```suggestion
- Se você ainda não tem uma conta, crie uma [visitando a página de sign-up](https://manager.azion.com/signup/).
```

2. Na página inicial, selecione a opção **Start with a template**.
3. Encontre o card do **Edge Function GitHub AutoDeploy** e selecione-o.
4. Clique na aba **Settings** para abrir o formulário de configuração.

---

> ## Configurar o template

Author: ```suggestion
## Configure o template
```

No formulário de configuração, você deve fornecer as informações para configurar sua nova aplicação, que incluirá a edge function. Preencha os campos apresentados.

Os campos identificados com asterisco são obrigatórios.

- **Application Name** *: o nome da sua aplicação. Ele é usado para identificar e gerenciar sua aplicação na plataforma da Azion.
   - Digite um nome único e fácil de lembrar. Se o nome já tiver sido usado, a plataforma retornará uma mensagem de erro.

> - **GitHub Personal Token** *: seu token pessoal do Github.

Reviewer: ```suggestion
- **GitHub Personal Token** *: seu token pessoal do GitHub.
```

:::tip
Ao gerar seu personal token do GitHub, conceda ao seu **scope** permissões para autorizar um aplicativo OAuth ou um personal token a acessar repositórios públicos e privados, incluindo acesso de leitura e reescrita de código. Você também deve habilitar a opção de **workflow** para permitir adicionar e atualizar arquivos de workflow no GitHub Actions.

> Observe que as permissões podem ter escopo para um usuário ou uma organização ou para um repositório. Leia a documentação sobre a [criação de um personal token no GitHub](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/managing-your-personal-access-tokens#creating-a-personal-access-token-classic) para obter mais informações.

Reviewer: ```suggestion
Observe que as permissões podem ter escopo para um usuário, para uma organização ou para um repositório. Leia a documentação sobre a [criação de um personal token no GitHub](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/managing-your-personal-access-tokens#creating-a-personal-access-token-classic) para obter mais informações.
```

> Observe que as permissões podem ter escopo para um usuário, para uma organização ou para um repositório. Leia a documentação sobre a [criação de um personal token no GitHub](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/managing-your-personal-access-tokens#creating-a-personal-access-token-classic) para obter mais informações.

Reviewer 2: ```suggestion
Observe que as permissões podem ter escopo para um usuário, para uma organização ou para um repositório. Leia a documentação sobre a [criação de um personal token no GitHub](https://docs.github.com/pt/authentication/keeping-your-account-and-data-secure/managing-your-personal-access-tokens#como-criar-um-fine-grained-personal-access-token) para obter mais informações.
```

---

## Implantar o template

Durante a implantação, você poderá acompanhar o processo através de uma janela mostrando os logs. Quando estiver concluída, uma nova página será exibida, confirmando que a edge application foi criada com sucesso.

Esta página mostra as seguintes seções:

- A seção **Access your Edge Application**, que inclui o link para visitar e explorar sua aplicação.
- O **Build Summary** contém o nome e a função da aplicação.
- A aba **Deployment details** pode ser aberta para acessar os logs relacionados à implantação.
- **What do you want to do next?** fornece recomendações sobre opções avançadas para aprimorar sua edge application: **Customize Domain**, **Manage Edge Application**, **View Edge Application Metrics**, **View Edge Application Logs** e o botão **Back to Home**.

:::note
O link para sua edge application permite que você veja como ela fica no navegador. No entanto, leva um certo tempo para propagar e configurá-la nas edge locations da Azion. Pode ser necessário aguardar alguns minutos para que a URL seja ativada e para que a página da aplicação seja efetivamente exibida no navegador.
:::

### Principais configurações

> A aplicação passa por um processo de criação otimizado, enfatizando no desempenho e a eficiência ideais, empregando os recursos da **Plataforma de Edge da Azion**.

Reviewer: ```suggestion
A aplicação passa por um processo de criação otimizado, enfatizando o desempenho e a eficiência ideais, empregando os recursos da **Plataforma de Edge da Azion**.
```

Após a implantação, o script cria um [repositório do GitHub](https://github.com/aziontech/azion-samples/tree/dev/templates/edge-function-github-autodeploy) para gerenciar o código-fonte e, em seguida, instala todas as dependências necessárias e várias configurações da edge application, o domínio para hospedar seu projeto e uma edge function para lógica personalizada.

O script também confirma os detalhes da implantação no repositório do GitHub, incluindo informações relacionadas à versão implantada e ao registro de hora e data da implantação.

> Se você quiser saber mais sobre como editar e personalizar as configurações do seu projeto, acesse a seção [Gerenciar o template](#managing-the-template).

Reviewer: ```suggestion
Se você quiser saber mais sobre como editar e personalizar as configurações do seu projeto, acesse a seção [Gerenciar o template](#gerenciar-o-template).
```

> Se você quiser saber mais sobre como editar e personalizar as configurações do seu projeto, acesse a seção [Gerenciar o template](#gerenciar-o-template).

Author: ```suggestion
Se você quiser saber mais sobre como editar e personalizar as configurações do seu projeto, acesse a seção [Gerenciar o template](#gerencie-o-template).
```

> Além disso, para garantir o desempenho ideal deste template, você deve ativar o seguinte produto da Azion:

Author: ```suggestion
Além disso, para garantir o desempenho ideal deste template, você deve ativar:
```

<Badge>
[Edge Functions](/pt-br/documentacao/produtos/edge-application/edge-functions/)
</Badge>

> Você deve ativar este produto separadamente via RTM. Para fazer isso:

Author: ```suggestion
Você deve ativá-lo separadamente via RTM. Para fazer isso:
```

1. [Acesse o RTM](https://manager.azion.com/).
2. No canto superior esquerdo, selecione as três linhas horizontais para abrir o **Products menu**.

> 3. Na seção **de** BUILD, selecione **Edge Application**.

Reviewer: ```suggestion
3. Na seção de **BUILD**, selecione **Edge Application**.
```

4. Você será redirecionado para a página de **Edge Application**.
- Ele lista todas as edge applications que você criou.
5. Encontre a aplicação relacionada ao seu template e selecione-a.
6. Na aba **Main Settings**, encontre a seção **Edge Application Modules** e ative a opção **Edge Functions**.

:::caution[Atenção]

> Se esse produto for ativado, a execução deste template poderá gerar custos relacionados ao uso. Consulte a [página de preços](https://www.azion.com/en/pricing/) para obter mais informações.

Reviewer: ```suggestion
Se esse produto for ativado, a execução deste template poderá gerar custos relacionados ao uso. Consulte a [página de preços](https://www.azion.com/pt-br/precos/) para obter mais informações.
```

:::

---

> ## Gerenciar o template

Author: ```suggestion
## Gerencie o template
```

Para gerenciar e editar as configurações de sua aplicação via Real-Time Manager (RTM), siga estas etapas:

1. [Acesse o RTM](https://manager.azion.com/).
2. No canto superior esquerdo, selecione as três linhas horizontais para abrir o **Products menu**.

> 3. Na seção **de** BUILD, selecione **Edge Application**.

Reviewer: ```suggestion
3. Na seção de **BUILD**, selecione **Edge Application**.
```

4. Você será redirecionado para a página de **Edge Application**.
- Ela lista todas as edge applications que você criou.

> 5. Encontre o edge application relacionada ao template **Edge Function GitHub AutoDeploy** e selecione-o.

Reviewer: ```suggestion
5. Encontre a edge application relacionada ao template **Edge Function GitHub AutoDeploy** e selecione-o.
```

- A lista é organizada em ordem alfabética. Você também pode usar a **barra de busca** localizada no canto superior esquerdo da lista; atualmente, ela é filtrada apenas pelo **Application Name**, ou nome da edge application.

Depois de selecionar a aplicação em que você trabalhará, você será direcionado para uma página que contém todas as configurações que você pode configurar.

:::tip
Leia a documentação sobre o [gerenciamento de edge applications](/en/documentation/products/edge-application/first-steps/) para obter mais detalhes.
:::

### Continuous deployment

Uma vez que o template é implantado, você pode editar e atualizar seus args e código na edge function, bem como implementar um fluxo de trabalho de continuous deployment, ou implantação contínua. No entanto, você precisará primeiro *declarar seus secrets no repositório do GitHub do seu projeto* para completar um segundo build com as alterações.

> Assim que esta segunda compilação for concluída, você poderá gerenciar seu projeto com um fluxo de trabalho de implantação contínua e editar os argumentos conforme desejado.

Reviewer: ```suggestion
Assim que essa segunda compilação for concluída, você poderá gerenciar seu projeto com um fluxo de trabalho de implantação contínua e editar os argumentos conforme desejado.
```

> Para fazer isso, abra seu novo repositório **Next.js Static Boilerplate** no GitHub. Em seguida, vá para **Settings** > **Secrets and Variables** > **Action** para [adicionar suas variáveis](https://docs.github.com/en/actions/security-guides/encrypted-secrets), seguindo estas instruções:

Reviewer: o nome do repositório deveria ser outro?

> Para fazer isso, abra seu novo repositório **Next.js Static Boilerplate** no GitHub. Em seguida, vá para **Settings** > **Secrets and Variables** > **Action** para [adicionar suas variáveis](https://docs.github.com/en/actions/security-guides/encrypted-secrets), seguindo estas instruções:

Author: ```suggestion
Para fazer isso, abra seu novo repositório gerado pelo deployment do template **Next.js Static Boilerplate** no GitHub. Em seguida, vá para **Settings** > **Secrets and Variables** > **Action** para [adicionar suas variáveis](https://docs.github.com/en/actions/security-guides/encrypted-secrets), seguindo estas instruções:
```

1. Adicione o Azion personal token aos *secrets*:

> - Leia [como gerar um personal token da Azion](/pt-br/documentacao/produtos/gestao-de-contas/personal-tokens/) na documentação.

Reviewer: dá pra linkar o guia!

> - Leia [como gerar um personal token da Azion](/pt-br/documentacao/produtos/gestao-de-contas/personal-tokens/) na documentação.

Author: ```suggestion
- Leia [como gerar um personal token da Azion](/pt-br/documentacao/produtos/guias/personal-tokens/) na documentação.
```

```bash
    AZION_PERSONAL_TOKEN=<value>
```

2. Adicione os ambientes para uso no action workflow no arquivo **main.yml**, dentro da pasta **.github/workflows** do seu repositório:

```yml title=".github/workflows/main.yml"
  - name: edge-...
    id: azion_edge
    ...
    with:
        ....
        azionPersonalToken: ${{ secrets.AZION_PERSONAL_TOKEN }}
        ....

```

3. Abra um pull request com as alterações na main branch e inicie a implantação automática.

Agora, seu projeto está pronto para trabalhar com um fluxo de trabalho de implantação contínua, atualizando instantaneamente quaisquer alterações no aplicativo ou no repositório.

> ### Adicionar um domínio personalizado

Author: ```suggestion
### Adicione um domínio personalizado
```

A edge application criada tem um domínio Azion atribuído para torná-la acessível através do navegador. O domínio tem o seguinte formato: `xxxxxxxxxx.map.azionedge.net`. No entanto, você pode adicionar um domínio personalizado para que os usuários acessem sua aplicação por meio dele.

import Button from '~/components/Button.astro';

> <Button href="/pt-br/documentacao/produtos/edge-application/domains/" text="leia a documentação de domínios" variant="secondary">

Author: ```suggestion
<Button href="/pt-br/documentacao/produtos/edge-application/domains/" text="saiba mais sobre domínios" variant="secondary">
```
